src/cli/completion.rs

```diff
@@ -0,0 +1,124 @@
+use crate::cli::{Args, CompletionCommand};
+use clap::CommandFactory;
+use miette::IntoDiagnostic;
+use regex::Regex;
+use std::borrow::Cow;
+use std::io::Write;
+use std::str::from_utf8_mut;
+
+pub(crate) fn execute(args: CompletionCommand) -> miette::Result<()> {
+    let clap_shell = args
+        .shell
+        .or(clap_complete::Shell::from_env())
+        .unwrap_or(clap_complete::Shell::Bash);
+
+    let mut script = vec![];
+
+    // Generate the original completion script.
+    clap_complete::generate(
+        clap_shell,
+        &mut Args::command(),
+        "pixi",
+        &mut script, // &mut std::io::stdout(),
+    );
+
+    match clap_shell {
+        clap_complete::Shell::Bash => {
+            let script = replace_bash_completion(from_utf8_mut(&mut script).into_diagnostic()?);
+            std::io::stdout()
+                .write_all(script.as_ref().as_ref())
+                .into_diagnostic()?;
+        }
+        _ => {
+            // If no replacements needed write original script to stdout
+            std::io::stdout().write_all(&script).into_diagnostic()?;
+        }
+    }
```

Comment:
Instead of having two code paths write to stdout, you could return the script from the match branches. E.g.

```rust
let script = match { ... }
stdout().write(script)
```

Reply:
Done that but is the `_ => Cow::Borrowed(from_utf8_mut(&mut script).into_diagnostic()?),` okay like that? It feels very wrong.